include/seadsa/AllocWrapInfo.hh

```diff
@@ -13,38 +13,39 @@
 namespace llvm {
 class Function;
 class Module;
-class Instruction;
 class TargetLibraryInfo;
 class TargetLibraryInfoWrapperPass;
 class Function;
 class Value;
 } // namespace llvm
 
 namespace seadsa {
-class AllocWrapInfo : public llvm::ModulePass {
+class AllocWrapInfo : public llvm::ImmutablePass {
 protected:
-  std::set<std::string> m_allocs;
-  std::set<std::string> m_deallocs;
+  mutable std::set<std::string> m_allocs;
+  mutable std::set<std::string> m_deallocs;
 
-  llvm::TargetLibraryInfoWrapperPass *m_tliWrapper;
+  mutable llvm::TargetLibraryInfoWrapperPass *m_tliWrapper;
 
-  void findAllocs(llvm::Module &M);
-  bool findWrappers(llvm::Module &M, std::set<std::string> &fn_names);
-  bool flowsFrom(llvm::Value *, llvm::Value *);
+  void findAllocs(llvm::Module &M) const;
```

Comment:
what is the reason for making all methods `const` and all fields `mutable`.
Is `ImmutablePass` in LLVM means `const`? I don't think so, but I might be wrong.

Reply:
`ImmutablePass` does not mean `const`. 
The reason for making methods `const` and fields `mutable` is that `AllocWrapInfo` object is always passed as `const&`. Its public methods are semantically `const` but after making the class immutable that are not syntactically `const` anymore. That's why I made the fields `mutable`.

Reply:
why is AllocWrapInfo passed as `const` when it is not `const` and when this is different from all other similar `WrapInfo` classes?

Reply:
This class identifies allocation wrappers and store them in some internal datastructures. Once this is computed an `AllocWrapInfo` object is read-only. So it makes sense to pass it as a `const` ref

Reply:
Once we made it immutable, things change. We can only identify the wrappers the first time a method to the class is called.

Reply:
ok